packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+	framework?: Framework;
+	packageManager?: PackageManagerAgent;
+	installProcess?: ExecaChildProcess;
+	userProfile?: PrismicUserProfile;
+	userRepositories?: PrismicRepository[];
+	repository?: {
+		domain: string;
+		exists: boolean;
+	};
+	projectInitialization?: {
+		patchedScript?: boolean;
+	};
+};
+
+export class SliceMachineInitProcess {
+	protected options: SliceMachineInitProcessOptions;
+	protected manager: SliceMachineManager;
+
+	protected context: SliceMachineInitProcessContext;
+
+	constructor(options?: SliceMachineInitProcessOptions) {
+		this.options = { ...DEFAULT_OPTIONS, ...options };
+		this.manager = createSliceMachineManager({ cwd: options?.cwd });
+
+		this.context = {};
+	}
+
+	async run(): Promise<void> {
+		// We prefer to manually allow console logs despite the app being a CLI to catch wild/unwanted console logs better
+		// eslint-disable-next-line no-console
+		console.log(
+			`\n${chalk.bgGray(` ${chalk.bold.white("Slice Machine")} `)} ${chalk.dim(
+				"→",
+			)} Init command started\n`,
+		);
+
+		await this.manager.telemetry.initTelemetry();
+		await this.manager.telemetry.track({
+			event: "command:init:start",
+			repository: this.options.repository,
+		});
+
+		try {
+			await this.detectEnvironment();
+
+			assertExists(
+				this.context.framework,
+				"Project framework must be available through context to proceed",
+			);
+
+			await this.beginCoreDependenciesInstallation();
+			await this.loginAndFetchUserData();
+
+			if (this.options.repository) {
+				await this.useRepositoryFlag();
+			} else {
+				await this.selectRepository();
+			}
+
+			assertExists(
+				this.context.repository,
+				"Repository selection must be available through context to proceed",
+			);
+
+			if (!this.context.repository.exists) {
+				await this.createNewRepository();
+			}
+
+			await this.finishCoreDependenciesInstallation();
+			await this.upsertSliceMachineConfigurationAndStartPluginRunner();
+			await this.pushDataToPrismic();
+			await this.initializeProject();
+			await this.initializePlugins();
+		} catch (error) {
+			await this.trackError(error);
+
+			throw error;
+		}
+
+		await this.manager.telemetry.track({
+			event: "command:init:end",
+			framework: this.context.framework.prismicName,
+			repository: this.context.repository.domain,
+			success: true,
+		});
+
+		// We prefer to manually allow console logs despite the app being a CLI to catch wild/unwanted console logs better
+		// eslint-disable-next-line no-console
+		console.log(
+			`\n${chalk.bgGreen(` ${chalk.bold.white("Slice Machine")} `)} ${chalk.dim(
+				"→",
+			)} Init command successful!`,
+		);
+
+		try {
+			// We prefer to manually allow console logs despite the app being a CLI to catch wild/unwanted console logs better
+			// eslint-disable-next-line no-console
+			console.log(`
+  YOUR REPOSITORY
+    Dashboard            ${chalk.cyan(
+			`https://${this.context.repository.domain}.prismic.io`,
```

Comment:
hmm, could this confuse people using the staged and dev environments?

Reply:
Yes, good catch. The correct URLs will be used according to the environment as of this commit: https://github.com/prismicio/slice-machine/commit/3f2f6f2336fc3f4505f48e04e426f04c949bf2a1